.github/workflows/release.yml

```diff
@@ -107,69 +105,51 @@ jobs:
+        uses: dbt-labs/dbt-adapters/.github/actions/build-hatch@bumpVersion
         with:
-          working-dir: "./dbt-tests-adapter/"
-
-      - name: Setup `hatch`
-        uses: ./.github/actions/setup-hatch
+          changelog-path: ${{ needs.bump-version-generate-changelog.outputs.changelog_path }}
+          artifacts-name: ${{ inputs.version_number }}-${{ inputs.package }}-${{ inputs.deploy-to }}
 
-      - name: Build `dbt-adapters`
-        if: ${{ inputs.package == 'dbt-adapters' }}
-        uses: ./.github/actions/build-hatch
-
-      - name: Build `dbt-tests-adapter`
+      - name: "Build `dbt-tests-adapter`"
         if: ${{ inputs.package == 'dbt-tests-adapter' }}
-        uses: ./.github/actions/build-hatch
+        uses: dbt-labs/dbt-adapters/.github/actions/build-hatch@bumpVersion
         with:
+          changelog-path: ${{ needs.bump-version-generate-changelog.outputs.changelog_path }}
           working-dir: "./dbt-tests-adapter/"
-
-      # this step is only needed for the release process
-      - name: "Upload Build Artifact"
-        if: ${{ github.event_name == 'workflow_call' }}
-        uses: actions/upload-artifact@v3
-        with:
-          name: ${{ steps.version.outputs.version_number }}
-          path: |
-            ${{ inputs.changelog_path }}
-            ./dist/
-          retention-days: 3
+          artifacts-name: ${{ inputs.version_number }}-${{ inputs.package }}-${{ inputs.deploy-to }}
 
   github-release:
-    name: GitHub Release
+    name: "GitHub Release"
     if: ${{ !failure() && !cancelled() }}
-
-    needs: [build-and-test]
-
-    uses: dbt-labs/dbt-release/.github/workflows/github-release.yml@main
-
+    needs: [build-and-test, bump-version-generate-changelog]
+    uses: dbt-labs/dbt-adapters/.github/workflows/github-release.yml@bumpVersion
     with:
+      sha: ${{ needs.bump-version-generate-changelog.outputs.final_sha }}
       version_number: ${{ inputs.version_number }}
       changelog_path: ${{ needs.bump-version-generate-changelog.outputs.changelog_path }}
       test_run: ${{ inputs.deploy-to == 'test' && true || false }}
-      sha: ${{ needs.bump-version-generate-changelog.outputs.final_sha }}
+      artifact_name: ${{ inputs.version_number }}-${{ inputs.package }}-${{ inputs.deploy-to }}
 
   pypi-release:
-    name: Publish to PyPI
+    name: "Publish to PyPI"
     runs-on: ubuntu-latest
     needs: [github-release]
     environment:
       name: ${{ inputs.deploy-to }}
       url: ${{ vars.PYPI_PROJECT_URL }}
     steps:
+      - name: "Check out repository"
+        uses: actions/checkout@v4
+
       - name: "Publish to PyPI"
-        uses: ./.github/actions/publish-pypi
+        uses: dbt-labs/dbt-adapters/.github/actions/publish-pypi@bumpVersion
         with:
           repository-url: ${{ vars.PYPI_REPOSITORY_URL }}
-          artifacts-dir: ${{ inputs.version_number }}
+          artifacts-name: ${{ inputs.version_number }}-${{ inputs.package }}-${{ inputs.deploy-to }}
```

Comment:
This gets calculated so many times. I feel like it should be in one spot to avoid falling out of sync if we change this name.